docs/src/main/asciidoc/extension-authors-guide.adoc

```diff
@@ -296,40 +296,141 @@ You will also need to configure the `maven-compiler-plugin` to detect the `quark
 
 === Build Step Processors
 
-Work is done at deployment time by producing and consuming instances of `io.quarkus.builder.item.BuildItem`. This is done
-by creating a class that has method(s) annotated with `io.quarkus.deployment.annotations.BuildStep`. These classes can
-consume items by injection, and produce items by either returning them from the method or by injecting an
-instance of `io.quarkus.deployment.annotations.BuildProducer` for the produced type. These processors can also record
-bytecode invocations, which is mapped to a `BuildItem` transparently.
+Work is done at augmentation time by _build steps_ which produce and consume _build items_.  The build steps found in
+the deployment modules that correspond to the extensions in the project build are automatically wired together and executed
+to produce the final build artifact(s).
```

Comment:
Can you add

Think of build items as a way for different extensions to communicate with one another, for example:

- expose the fact that a database configuration exists
- consume that database configuration (e.g. a connection pool extension or an ORM extension)
- ask an extension to do work for another extension: e.g. an extension wanting to define a new CDI bean and asking the ArC extension

This is a very flexible mechanism.